mc_test/mcTestCall.cpp

```diff
@@ -1,583 +1,653 @@
- *  "ft_min_amount" "num_constraints"
- *  "bt_list_len", "pk_dest_0" "amount_0" "pk_dest_1" "amount_1" ... "pk_dest_n" "amount_n",
- *  "custom_fields_list_len", "custom_field_0", ... , "custom_field_1"
- *  Generates a TestCertificateProof.
- *  NOTE: "constant" param must be present if "cert" has been passed; If "cert_no_const" has been passed,
- *        instead, "constant" param must not be present.
- *
- *  3) ./mcTest "create" "csw/csw_no_const" "darlin/cob_marlin" <"-v"> <"-zk"> "proof_path" "params_dir" "segment_size",
- *  "amount" "sc_id" "nullifier" "mc_pk_hash" "end_cum_comm_tree_root" "num_constraints" <"cert_data_hash">, [constant]
- *  Generates a TestCSWProof. cert_data_hash is optional.
- *  NOTE: "constant" param must be present if "cert" has been passed; If "cert_no_const" has been passed,
- *        instead, "constant" param must not be present.
- */
-
-void create_verify_test_cert_proof(std::string ps_type_raw, std::string cert_type_raw, int argc, char** argv) {
-    int arg = 4;
-    bool verify = false;
-    if (std::string(argv[arg]) == "-v"){
-        arg++;
-        verify = true;
-    }
-
-    bool zk = false;
-    if (std::string(argv[arg]) == "-zk") {
-        arg++;
-        zk = true;
-    }
-
-    // Get ProvingSystemType
-    ProvingSystem ps_type;
-    if (ps_type_raw == "darlin") {
-        ps_type = ProvingSystem::Darlin;
-    } else if (ps_type_raw == "cob_marlin") {
-        ps_type = ProvingSystem::CoboundaryMarlin;
-    } else {
-        abort(); // Invalid ProvingSystemType
-    }
-
-    // Parse inputs
-    // Parse paths
-    auto proof_path = std::string(argv[arg++]);
-    size_t proof_path_len = proof_path.size();
+#include <unistd.h>
+#include <cstring>
+#include <sstream>
+
+char const* usage = R"(
+    OPERATION [OPTIONS...] params_directory [CREATE_PARAMETERS... (CERT_PAR/CSW_PAR)]
+
+    OPERATION:
+    generate         generates SNARK pk and vk for a test
+    create           creates a TestCertificateProof/TestCSWProof.
+
+    OPTIONS:
+    -c circuit       circuit type {cert, csw} (default: cert)
+    -k constant      constant field element (default: no constant)
+    -p ps            proving system type {darlin, cob_marlin} (default: cob_marlin)
+    -s segsize       segment size (default: 512)
+    -n constraints   number of constraints (default: 1024)
+    -r               use keyrotation (default: false)
+    -v               verify the proof created (default: false)
+    -z               use zero knowledge (default: false)
+
+    CREATE_PARAMETERS (must be given in the exact order):
+    output_proof_file sc_id end_cum_comm_tree_root cert_datahash
+
+    CERT_PAR
+    epoch_number quality btr_fee ft_min_amount bt_list_len mc_dest_addr_0 amount_0 mc_dest_addr_1 amount_1 mc_dest_addr_n amount_n custom_fields_list_len custom_field_0 custom_field_1
+
+    CSW_PAR
+    nullifier mc_address
+
+    Notes: cert_datahash serves different purposes in different contexts.
```

Comment:
We can call it simply prev_cert_data_hash: in both cases it's indeed the hash of the previous certificate